src/msg/Message.h

```diff
@@ -519,4 +519,13 @@ extern void encode_message(Message *m, uint64_t features, bufferlist& bl);
 extern Message *decode_message(CephContext *cct, int crcflags,
                                bufferlist::const_iterator& bl);
 
+template <class MessageType>
+class MessageFactory {
```

Comment:
End goal is to make Message constructors private/protected so that it's not possible to create them except through the factory. This pattern requires that the constructors remain public.